Tehtris/tehtris_modules/trigger_tehtris_events.py

```diff
@@ -77,15 +80,34 @@ def __fetch_next_events(self, from_date: datetime, offset: int):
             )
             return events
 
+    def _remove_duplicates(self, events: list[dict[str, Any]]) -> list[dict[str, Any]]:
+        """
+        Remove duplicates events from the fetched events and update the cache with new ids.
+
+        Args:
+            events: list[dict[str, Any]]
+
+        Returns:
+            list[dict[str, Any]]:
+        """
+        result = [event for event in events if event["uid"] not in self.events_cache]
+        self.events_cache.clear()
```

Comment:
Do we need to clear the cache? As an LRU cache, it should manage the key itself.